test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/PrimitiveTypesTests/Tests/PrimitiveValueFormatTest.cs

```diff
@@ -0,0 +1,476 @@
+
+namespace Microsoft.OData.Client.E2E.Tests.PrimitiveTypesTests.Tests;
+
+public class PrimitiveValueFormatTest : EndToEndTestBase<PrimitiveValueFormatTest.TestsStartup>
+{
+    private readonly Uri _baseUri;
+    private readonly Container _context;
+    private readonly IEdmModel _model;
+
+    public class TestsStartup : TestStartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.ConfigureControllers(typeof(PrimitiveKeyValuesTestsController), typeof(MetadataController));
+
+            services.AddControllers().AddOData(opt =>
+                opt.EnableQueryFeatures().AddRouteComponents("odata", PrimitiveKeyValuesEdmModel.GetEdmModel()));
+        }
+    }
+
+    public PrimitiveValueFormatTest(TestWebApplicationFactory<PrimitiveValueFormatTest.TestsStartup> fixture) 
+        : base(fixture)
+    {
+        if (Client.BaseAddress == null)
+        {
+            throw new ArgumentNullException(nameof(Client.BaseAddress), "Base address cannot be null");
+        }
+
+        _baseUri = new Uri(Client.BaseAddress, "odata/");
+
+        _context = new Container(_baseUri)
+        {
+            HttpClientFactory = HttpClientFactory
+        };
+
+        _model = PrimitiveKeyValuesEdmModel.GetEdmModel();
+        ResetDefaultDataSource();
+    }
+
+    public static IEnumerable<object[]> MimeTypesData
+    {
+        get
+        {
+            yield return new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterFullMetadata };
+            yield return new object[] { MimeTypes.ApplicationJson + MimeTypes.ODataParameterFullMetadata + MimeTypes.ODataParameterIEEE754Compatible };
+        }
+    }
+
+    public string DataPattern = "(" + @"[\+-]?\d\.?\d*E?[\+-]?\d*" + "|" + @"INF|-INF|NaN" + ")";
+
+    [Theory]
+    [MemberData(nameof(MimeTypesData))]
+    public async Task LongWithoutSuffixAsKeyInURLTest(string mimeType)
+    {
+        // Arrange
+        var expectedKeySegment = "EdmInt64Set(1)";
+
+        // Act & Assert
+        await PrimitiveValueAsKeyInURLAsync("EdmInt64Set(1)", mimeType, expectedKeySegment);
+    }
+
+    [Theory]
+    [MemberData(nameof(MimeTypesData))]
+    public async Task LongWithSuffixAsKeyInURLTest(string mimeType)
+    {
+        // Arrange
+        var expectedKeySegment = "EdmSingleSet(1)";
+
+        // Act & Assert
+        await PrimitiveValueAsKeyInURLAsync("EdmSingleSet(1L)", mimeType, expectedKeySegment);
+        await PrimitiveValueAsKeyInURLAsync("EdmSingleSet(1l)", mimeType, expectedKeySegment);
+    }
+
+    [Theory]
+    [MemberData(nameof(MimeTypesData))]
+    public async Task FloatWithoutSuffixAsKeyInURLTest(string mimeType)
+    {
+        // Arrange
+        var expectedKeySegment = "EdmSingleSet(1)";
+
+        // Act & Assert
+        await PrimitiveValueAsKeyInURLAsync("EdmSingleSet(1.0)", mimeType, expectedKeySegment);

```

Comment:
It's not clear what this method is doing. Is it possible to write the relevant "act" and "assert" portions directly?